src/main/java/ch/uzh/ifi/hase/soprafs23/controller/UserController.java

```diff
@@ -346,4 +347,31 @@ public ResponseEntity<Void> setReady(@PathVariable Long userId, @PathVariable Lo
 
         return new ResponseEntity<>(HttpStatus.OK); //TODO returns 200 or 204?
     }
+
+    @GetMapping("/users/{userId}/{groupId}/ready")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public Map<Long, Boolean> getGroupUserReady(@PathVariable Long userId, @PathVariable Long groupId, HttpServletRequest request) {
+
+        // 404 - user not found
+        User user = userService.getUserById(userId);
+
+        // 404 - group not found
+        Group group = groupService.getGroupById(groupId);
+
+        // 401 - not authorized if not the host
+        Long tokenId = userService.getUseridByToken(request.getHeader("X-Token"));
+        if(!tokenId.equals(userId) || !userId.equals(group.getHostId())) {
+            throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, "You are not authorized to view this information.");
+        }
+
+        // 401 - not authorized if not a member of the group
+        List<Long> memberIds = groupService.getAllMemberIdsOfGroup(group);
+        if(!memberIds.contains(tokenId)) {
+            throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, String.format("You are not a member of the group with id %d.", groupId));
+        }
```

Comment:
I think this check is not necessary.. it would never fall into this if, since if the user isn't a member, he was also not the host, and then the previous 401 was thrown..

but is that how this is intended? should guests not be able to see the status of the others?